app/_src/policies/meshexternalservice.md

```diff
@@ -0,0 +1,282 @@
+{% endwarning %}
+
+This resource allows services running inside the mesh to consume services that are not part of the mesh.
+The `MeshExternalService` resource allows you to declare external resources instead of relying on [MeshPassthrough](/docs/{{ page.version }}/policies/meshpassthrough) or [passthrough mode](/docs/{{ page.version }}/networking/non-mesh-traffic#outgoing).
+In contrast to passthrough `MeshExternalService` behaves like a normal service as if it was part of the mesh.
+
+{% warning %}
+Currently `MeshExternalService` resource does not support targeting by [targetRef policies](/docs/{{ page.version }}/policies/targetref).
+This limitation will be lifted in the next release.
+{% endwarning %}
+
+## Configuration
+
+### Match
+
+This section specifies the rules for matching traffic that will be routed to external resources defined in `endpoints` section.
+The only `type` supported is [HostnameGenerator](/docs/{{ page.version }}/policies/hostnamegenerator) and it means that it will match traffic directed to a hostname created by the hostname generator.
+The `port` field when omitted means that all traffic will be matched.
+Protocols that are supported are: `tcp`, `grpc`, `http`, `http2`.
+
+```yaml
+match:
+  type: HostnameGenerator
+  port: 4244
+  protocol: tcp
+```
+
+### Endpoints
+
+This section specifies the destination of the matched traffic.
+It's possible to define IPs, DNS names and unix domain sockets.
+
+```yaml
+endpoints:
+  - address: 1.1.1.1
+    port: 12345
+  - address: example.com
+    port: 80
+  - address: unix:///tmp/example.sock
+```
+
+### TLS
+
+This section describes the TLS and verification behaviour.
+You can define TLS version requirements, option to allow renegotiation, verification of SNI, SAN, custom CA and client certificate and key for server verification.
+To disable parts of the verification you can set different `mode` - `SkipSAN`, `SkipCA`, `SkipAll`, `Secured` (default).
+
+```yaml
+tls:
+  version:
+    min: TLS12
+    max: TLS13
+  allowRenegotiation: false
+  verification:
+    mode: SkipCA
+    serverName: "example.com"
+    subjectAltNames:
+      - type: Exact
+        value: example.com
+      - type: Prefix
+        value: "spiffe://example.local/ns/local"
+    caCert:
+      inline: dGVzdA==
+    clientCert:
+      secret: "123"
+    clientKey:
+      secret: "123"
+```
+
+When TLS is enabled but the CA is not set, the sidecar uses the default OS-specific CA. The user can override the default CA by setting the path in the environment variable `KUMA_DATAPLANE_RUNTIME_DYNAMIC_SYSTEM_CA_PATH` for the sidecar
+
+### DNS setup
+
+To be able to access `MeshExternalService` via a hostname you need to define a [HostnameGenerator](/docs/{{ page.version }}/policies/hostnamegenerator) with a `meshExternalService` selector.
+
+## Examples
+
+For the following examples the following `HostnameGenerator` will be used:
+
+{% policy_yaml hostnamegenerator %}
```

Comment:
I think we should add explanations to examples

Reply:
added